inc/zoo/map/RobinHood.h

```diff
@@ -387,62 +336,89 @@ struct RH_Frontend_WithSkarupkeTail {
         auto swarIndex = index / MD::Lanes;
         auto intraIndex = index % MD::Lanes;
         auto mdp = this->md_.data() + swarIndex;
-        constexpr auto MaxRelocations = 64;
+        constexpr auto MaxRelocations = 100;
         std::array<std::size_t, MaxRelocations> relocations;
+        std::array<int, MaxRelocations> newElements;
         auto relocationsCount = 0;
-        auto elementToInsert = MD(needle.at(intraIndex));
+        auto elementToInsert = needle.at(intraIndex);
+
+        // The very last element in the metadata will always have a psl of 0
+        // this serves as a sentinel for insertions, the only place to make
+        // sure the table has not been exhausted is an eviction chain that
+        // ends in the sentinel
+        // Also, the encoding for the PSL may be exhausted
         for(;;) {
             // Loop invariant:
             // deadline, index, swarIndex, intraIndex, elementToInsert correct
             // mdp points to the haystack that gave the deadline
-
-            // back this up to pick the element being evicted, if there is one
-            auto mdBackup = *mdp;
-            auto evictedPSL = mdBackup.PSLs().at(intraIndex);
+            auto md = *mdp;
+            auto evictedPSL = md.PSLs().at(intraIndex);
             if(0 == evictedPSL) { // end of eviction chain!
-                assignMetadataElement(
-                    deadline,
-                    elementToInsert.shiftLanesLeft(intraIndex),
-                    mdp
-                );
+                if(SlotCount - 1 <= index) {
```

Comment:
Name unclear: is index here a global index, a swar index, or a local offset?

Reply:
within the code, I attempt to use these consistently:
`index`: the global index of elements
`swarIndex`,
`intraIndex`